docs/frameworks/sveltekit.md

```diff
@@ -4,212 +4,252 @@ title: SvelteKit | Frameworks
-            e.url = '/'
-          else if (url.endsWith('index.html'))
-            e.url = `/${url.substring(0, url.lastIndexOf('/'))}`
-          else if (url.endsWith('.html'))
-            e.url = `/${url.substring(0, url.length - '.html'.length)}`
-
-        }
-
-        return e
-      })
-
-      return { manifest }
-    }]
-  }
-}
+::: details $lib/ReloadPrompt.svelte
+```html
+<script lang="ts">
+	import { useRegisterSW } from 'virtual:pwa-register/svelte'
+	const {
+		needRefresh,
+		updateServiceWorker
+	} = useRegisterSW({
+		onRegistered(r) {
+		// uncomment following code if you want check for updates
+		// r && setInterval(() => {
+		//    console.log('Checking for sw update')
+		//    r.update()
+		// }, 20000 /* 20s for testing purposes */)
+			console.log(`SW Registered: ${r}`)
+		},
+		onRegisterError(error) {
+			console.log('SW registration error', error)
+		},
+	})
+	const close = () => {
+		offlineReady.set(false)
+		needRefresh.set(false)
+	}
+	$: toast = $offlineReady || $needRefresh
+</script>
 
-export { pwaConfiguration }
-```
-:::
+{#if toast}
+	<div class="pwa-toast" role="alert">
+		<div class="message">
+			{#if $offlineReady}
+				<span>
+					App ready to work offline
+				</span>
+			{:else}
+				<span>
+					New content available, click on reload button to update.
+				</span>
+			{/if}
+		</div>
+		{#if $needRefresh}
+			<button on:click={() => updateServiceWorker(true)}>
+				Reload
+			</button>
+		{/if}
+		<button on:click={close}>
+			Close
+		</button>
+	</div>
+{/if}
 
-::: details 3) modify your build script
-```json
-{
-  "scripts": {
-    "build": "svelte-kit build && node ./pwa.js"
-  }
-}
+<style>
+	.pwa-toast {
+		position: fixed;
+		right: 0;
+		bottom: 0;
+		margin: 16px;
+		padding: 12px;
+		border: 1px solid #8885;
+		border-radius: 4px;
+		z-index: 2;
+		text-align: left;
+		box-shadow: 3px 4px 5px 0 #8885;
+		background-color: white;
+	}
+	.pwa-toast .message {
+		margin-bottom: 8px;
+	}
+	.pwa-toast button {
+		border: 1px solid #8885;
+		outline: none;
+		margin-right: 5px;
+		border-radius: 2px;
+		padding: 3px 10px;
+	}
+</style>
 ```
 :::
 
-::: details 4) add Vite Plugin PWA to svelte.config.js
-```js
-import adapter from '@sveltejs/adapter-static'
-import preprocess from 'svelte-preprocess'
-import { VitePWA } from 'vite-plugin-pwa'
-import { pwaConfiguration } from './pwa-configuration.js'
-
-/** @type {import('@sveltejs/kit').Config} */
-const config = {
-  // Consult https://github.com/sveltejs/svelte-preprocess
-  // for more information about preprocessors
-  preprocess: preprocess(),
-
-  kit: {
-    adapter: adapter(),
-
-    // hydrate the <div id="svelte"> element in src/app.html
-    target: '#svelte',
-    vite: {
-      plugins: [VitePWA(pwaConfiguration)]
-    }
-  }
-}
+## SvelteKit Adapters
 
-export default config
-```
-:::
+If you use some feature in your SvelteKit Adapter, you should also configure the PWA plugin properly using the `kit` option:
```

Comment:
These mostly aren't adapter options, but just SvelteKit options. It'd be nice to make the SvelteKit PWA integration read the `svelte.config.js` so that users don't need to set the options a second time
```suggestion
If you set certain SvelteKit options, you should also configure the PWA plugin properly using the `kit` option:
```